merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/ResultsCellRepository.java

```diff
@@ -2,15 +2,18 @@
 
 import gov.nasa.jpl.aerie.merlin.server.ResultsProtocol;
 import gov.nasa.jpl.aerie.merlin.server.models.PlanId;
+import gov.nasa.jpl.aerie.merlin.server.models.SimulationDatasetId;
 
 import java.util.Optional;
 
 public interface ResultsCellRepository {
   ResultsProtocol.OwnerRole allocate(PlanId planId);
 
-  Optional<ResultsProtocol.OwnerRole> claim(PlanId planId, Long datasetId);
+  Optional<ResultsProtocol.OwnerRole> claim(PlanId planId, SimulationDatasetId datasetId);
 
   Optional<ResultsProtocol.ReaderRole> lookup(PlanId planId);
 
+  Optional<ResultsProtocol.ReaderRole> lookup(SimulationDatasetId datasetId);
```

Comment:
Following up on our discussion in slack, since simulation dataset ids are not the same thing as dataset ids, we should use the more verbose `simulationDatasetId` to be uber-clear.

When a lot of this code was written, the simulation dataset did not have a synthetic key.

Reply:
Done (: